### PR TITLE
WT-9226 Process the call log for begin_transaction()

### PR DESCRIPTION
**Pull request by Sean04**
I've added a boolean to the session class to indicate whether a transaction is running. Let me know if there's anything else we need to do for begin_transaction.

### CODE REVIEW

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (etienneptl) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:155]: Be careful with the other PR in parallel regarding this check

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (etienneptl) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:159]: Multi-line comment

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (etienneptl) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:164]: Do we want to simply break here or exit?

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (Sean04) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp:155]: Good point, I'll wait for the other PR to merge first and remove that check here.

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (Sean04) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp:164]: I guess we could just exit here to save other errors being printed if we continue to try doing other operations on an invalid session. I can change close session to reflect this too.

Comment (Sean04) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp:164]: By exit do you mean throw an error?

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (Sean04) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp]: Fixed.

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (Sean04) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp]: Fixed.

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (Siddhartha8899) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:177]: I think we should have a separate function called `begin_transaction()` in the session class. 
Note: Call log manager is an application of the WiredTiger simulator.

In that function we could do the following:
- Check if the transaction is running.
- If yes, return false (or error out) 
- set the transaction_running = true

Comment (Sean04) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp:177]: Ah okay, that makes sense; leave the responsibility to the session.

Comment (Siddhartha8899) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp:177]: Yup.

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (etienneptl) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:165]: I would probably return a `bool` instead.

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (etienneptl) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:166]: I would move this err log in the `session_simulator`. The caller may not know why we could not begin the transaction. In the future, we could prevent from starting a new transaction for other reasons if that makes sense.

File: test/simulator/timestamp/src/session_simulator.cpp
Comment (etienneptl) [inline test/simulator/timestamp/src/session_simulator.cpp:45]: This comment does not seem correct. We are not interacting with the map.

File: test/simulator/timestamp/src/session_simulator.cpp
Comment (Siddhartha8899) [inline test/simulator/timestamp/src/session_simulator.cpp:40]: How about?
`Cannot begin a new transaction as there is an existing transaction that is currently running. `

**Review by Siddhartha8899 — CHANGES_REQUESTED** [review]

File: test/simulator/timestamp/src/session_simulator.cpp
Comment (Sean04) [reply test/simulator/timestamp/src/session_simulator.cpp]: Fixed.

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (Sean04) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp]: That's true.

File: test/simulator/timestamp/src/session_simulator.cpp
Comment (Siddhartha8899) [inline test/simulator/timestamp/src/session_simulator.cpp:39]: I think `s_all` did this (multi-line comment)?

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (Siddhartha8899) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:166]: We should break outside the if statement too?

**Review by Siddhartha8899 — CHANGES_REQUESTED** [review]

File: test/simulator/timestamp/src/session_simulator.cpp
Comment (Sean04) [reply test/simulator/timestamp/src/session_simulator.cpp:39]: The comment tags don't fit on the same line.

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (Sean04) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp:166]: Ah true, I guess we can just leave this if then and handle it when we figure out error handling?

Comment (Siddhartha8899) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp:166]: yup makes sense.

File: test/simulator/timestamp/src/session_simulator.cpp
Comment (Siddhartha8899) [reply test/simulator/timestamp/src/session_simulator.cpp:39]: Yea, best to make the comment multi-line then.

File: test/simulator/timestamp/call_log_manager/call_log_manager.h
Comment (Siddhartha8899) [inline test/simulator/timestamp/call_log_manager/call_log_manager.h:37]: You forgot to initialise begin_transaction in the map.

**Review by Siddhartha8899 — CHANGES_REQUESTED** [review]

File: test/simulator/timestamp/call_log_manager/call_log_manager.h
Comment (Sean04) [reply test/simulator/timestamp/call_log_manager/call_log_manager.h:37]: Thanks

**Review by Siddhartha8899 — APPROVED** [review]
LGTM, please merge the pm branch (after WT-9218) before merging this PR.

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (etienneptl) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:126]: np: Can we have all the case statements in the A-Z order?

File: test/simulator/timestamp/src/include/session_simulator.h
Comment (etienneptl) [inline test/simulator/timestamp/src/include/session_simulator.h:34]: Do we need the default constructor/destructor? Since we don't do anything specific with them, we probably don't need those lines.

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (Sean04) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp:126]: Yeah, can do

File: test/simulator/timestamp/src/include/session_simulator.h
Comment (Sean04) [reply test/simulator/timestamp/src/include/session_simulator.h:34]: I guess so, I'm not sure if we plan on using them.

Comment (Siddhartha8899) [reply test/simulator/timestamp/src/include/session_simulator.h:34]: We need the constructor, we use for conn->open_session.

**Review by etienneptl — APPROVED** [review]
LGTM!

**Pull request merged**
